config/boards/xiaomi-lmi.wip

```diff
@@ -0,0 +1,106 @@
+# Generate kernel and rootfs image for Qcom ABL booting
+declare -g BOARD_NAME="Xiaomi lmi"
+declare -g BOARD_MAINTAINER="FantasyGmm"
+declare -g BOARDFAMILY="qcom-abl"
+declare -g KERNEL_TARGET="sm8250"
+declare -g KERNELPATCHDIR="arm64-sm8250"
+declare -g EXTRAWIFI="no"
+declare -g MODULES="spi-geni-qcom"
+declare -g BOOTCONFIG="none"
+declare -g IMAGE_PARTITION_TABLE="gpt"
+declare -g -a ABL_DTB_LIST=("sm8250-xiaomi-lmi")
+
+# Use the full firmware, complete linux-firmware plus Armbian's (for qcom/a650_sqe.fw)
+declare -g BOARD_FIRMWARE_INSTALL="-full"
+
+function post_family_config_branch_sm8250__pmos_kernel() {
+	display_alert "Setting up kernel for" "${BOARD}" "info"
+	declare -g KERNEL_MAJOR_MINOR="6.7" # Major and minor versions of this kernel.
+	declare -g KERNELBRANCH='branch:linux-6.7.y'
+}
+
+function xiaomi-lmi_is_userspace_supported() {
+	[[ "${RELEASE}" == "jammy" ]] && return 0
+	[[ "${RELEASE}" == "trixie" ]] && return 0
+	[[ "${RELEASE}" == "noble" ]] && return 0
+	return 1
+}
+
+function post_family_tweaks_bsp__xiaomi-lmi_firmware() {
+	display_alert "$BOARD" "Install firmwares for xiaomi lmi" "info"
+
+	# Alsa-ucm-conf profile for Xiaomi lmi
+	mkdir -p $destination/usr/share/alsa/ucm2/conf.d/sm8250
+	install -Dm644 $SRC/packages/bsp/xiaomi-lmi/lmi.conf $destination/usr/share/alsa/ucm2/Xiaomi/lmi/lmi.conf
+	install -Dm644 $SRC/packages/bsp/xiaomi-lmi/lmi_HiFi.conf $destination/usr/share/alsa/ucm2/Xiaomi/lmi/HiFi.conf
+	ln -sfv ../../Xiaomi/lmi/lmi.conf \
+		"$destination/usr/share/alsa/ucm2/conf.d/sm8250/Xiaomi Mi 10.conf"
+
+	# USB Gadget Network service
+	mkdir -p $destination/usr/local/bin/
+	mkdir -p $destination/usr/lib/systemd/system/
+	install -Dm655 $SRC/packages/bsp/usb-gadget-network/setup-usbgadget-network.sh $destination/usr/local/bin/
+	install -Dm655 $SRC/packages/bsp/usb-gadget-network/remove-usbgadget-network.sh $destination/usr/local/bin/
+	install -Dm644 $SRC/packages/bsp/usb-gadget-network/usbgadget-rndis.service $destination/usr/lib/systemd/system/
+
+	# Kernel postinst script to update abl boot partition
+	install -Dm655 $SRC/packages/bsp/xiaomi-lmi/zz-update-abl-kernel $destination/etc/kernel/postinst.d/
+
+	return 0
+}
+
+function post_family_tweaks__xiaomi-lmi_enable_services() {
+	if ! xiaomi-lmi_is_userspace_supported; then
```

Comment:
I wonder if you should fail early and not proceed with the build if userspace is not supported. Something like - https://github.com/armbian/build/blob/6fa5981348edadb298acc0e742262976e9124050/config/sources/families/bcm2711.conf#L78

Reply:
Is there a reason to proceed with the build even when the required software is not available?

Reply:
If these packages are  not installed, kernel would still boot, but with some driver/system limitation. Although we don't recommand people building unsupported release, someone may need to build one unsupported image to do something he like, for example, add support to it.

Reply:
ok...was just curious about it